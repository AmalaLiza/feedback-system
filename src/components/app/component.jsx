import React from "react";
import { Switch, Route } from "react-router-dom";
import Results from "../../pages/results/container";
import Home from "../../pages/home/container";
import "../../global.css";

const App = () => {
	/*
		A <Switch> looks through all its children <Route>
		elements and renders the first one whose path
		matches the current URL. Use a <Switch> any time
		you have multiple routes, but you want only one
		of them to render at a time
		*/
	return (
		<Switch>
			<Route path="/" exact component={Home} />
			<Route path="/results" component={Results} />
		</Switch>
	);
};
export default App;
