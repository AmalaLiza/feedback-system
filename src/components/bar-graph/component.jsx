import React from "react";
import PropTypes from "prop-types";
import { BarChart, Bar, XAxis, YAxis, CartesianGrid, Tooltip } from "recharts";
import styles from "./styles.css";

/**
 * CustomTooltip component for charts.
 */
const CustomTooltip = ({ payload = {}, xAxisDataKey, barDataKey, active }) => {
	if (active) {
		return (
			<>
				<p className={styles.label}>
					{`${xAxisDataKey}: ${payload[0]?.payload?.[xAxisDataKey]}`}
				</p>
				<p className={styles.label}>
					{`${barDataKey}: ${payload[0]?.payload?.[barDataKey]}`}
				</p>
			</>
		);
	}
	return null;
};

/**
 * Chart component
 * Props are chartData, xAxisDataKey, barDataKey
 */
const BarGraph = ({ chartData, xAxisDataKey, barDataKey }) => (
	<BarChart
		width={500}
		height={400}
		data={chartData}
		fillOpacity={1}
		margin={{
			top: 10,
			right: 30,
			left: 0,
			bottom: 0,
		}}>
		<CartesianGrid strokeDasharray="3 3" />
		<XAxis dataKey={xAxisDataKey} />
		<YAxis ticks={[0, 1, 2, 3, 4, 5]} domain={[0, 5]} />
		<Tooltip
			content={
				<CustomTooltip barDataKey={barDataKey} xAxisDataKey={xAxisDataKey} />
			}
		/>
		<Bar
			fillOpacity={0.8}
			type="monotone"
			dataKey={barDataKey}
			stroke="#f15c41"
			fill="url(#colorUv)"
		/>
		<defs>
			<linearGradient id="colorUv" x1="0" y1="0" x2="0" y2="1">
				<stop offset="5%" stopColor="#e052a0" stopOpacity={1} />
				<stop
					offset="95%"
					stopColor="#f15c41"
					stopOpacity={1}
					fillOpacity={1}
				/>
			</linearGradient>
		</defs>
	</BarChart>
);

BarGraph.propTypes = {
	/**
	 * chartData: Data to be rendered in chart.
	 */
	chartData: PropTypes.array.isRequired,
	/**
	 * xAxisDataKey: dataKey to be plotted on x-axis.
	 */
	xAxisDataKey: PropTypes.string.isRequired,
	/**
	 * barDataKey: dataKey to be plotted as bars.
	 */
	barDataKey: PropTypes.string.isRequired,
};

export default BarGraph;
