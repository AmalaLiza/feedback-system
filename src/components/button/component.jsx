import React, { Component } from "react";
import styles from "./styles.js";

class Button extends React.Component {
	constructor(props) {
		super(props);
		this.state = { value: "" };
		this.handleClick = this.handleClick.bind(this);
	}

	handleClick(event) {
		console.log("on click");
	}

	render() {
		const { buttonText } = this.props;
		return <button onChange={this.handleClick}> {buttonText}</button>;
	}
}
export default Button;
