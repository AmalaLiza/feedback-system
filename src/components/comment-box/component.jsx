import React from "react";
import PropTypes from "prop-types";
import styles from "./styles.css";
import star from "../../images/star.svg";

/**
 * CommentBox component
 * Props are comment object
 */
const CommentBox = ({ comment }) => (
	<li className={styles.commentSection}>
		<header className={styles.header}>
			<div className={styles.avatar}>{comment.name[0]}</div>
			<span className={styles.name}>{comment.name}</span>
			<div className={styles.metadata}>
				<div className={styles.commentDate}>
					{new Date(comment.commentDate).toDateString()}
				</div>
				<div className={styles.commentDate}>
					<img className={styles.star} src={star} />
					{comment.rating}
				</div>
			</div>
		</header>
		<div className={styles.commentContent}>
			<p className={styles.email}>
				<span className={styles.emailLabel}>E-mail </span>
				<span>{comment.email}</span>
			</p>
			<p>{comment.comment}</p>
		</div>
	</li>
);

CommentBox.propTypes = {
	/**
	 * comment: comment object to be rendered.
	 */
	comment: PropTypes.shape({
		email: PropTypes.string,
		name: PropTypes.string,
		comment: PropTypes.string,
		rating: PropTypes.number,
		commentDate: PropTypes.number,
	}).isRequired,
};

export default CommentBox;
