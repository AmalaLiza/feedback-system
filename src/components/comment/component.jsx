import React from "react";
import styles from "./styles.css";
import star from "../../images/star.svg";

const CommentBox = ({ comment }) => (
	<li className={styles.commentSection}>
		<header className={styles.header}>
			<div className={styles.avatar}>{comment.name[0]}</div>
			<span className={styles.name}>{comment.name}</span>
			<div className={styles.metadata}>
				<div className={styles.commentDate}>
					{new Date(comment.comment_date).toDateString()}
				</div>
				<div className={styles.commentDate}>
					<img className={styles.star} src={star} />
					{comment.rating}
				</div>
			</div>
		</header>
		<div className={styles.commentContent}>
			<p className={styles.email}>
				<span className={styles.emailLabel}>E-mail </span>
				<span>{comment.email}</span>
			</p>
			<p>{comment.comment}</p>
		</div>
	</li>
);
export default CommentBox;
