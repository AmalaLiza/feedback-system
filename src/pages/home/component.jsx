import React, { Component } from "react";
import styles from "./styles.js";
import Button from "../../components/button/component";
import Input from "../../components/input/component";
import { useHistory, withRouter } from "react-router-dom";
import StarRating from "../../components/rating/component";

function HomeButton() {
	let history = useHistory();

	function onSubmit() {
		history.push("/results");
	}

	return <Button buttonText="buttonText" handleClick={onSubmit} />;
}

class Home extends Component {
	constructor(props) {
		super(props);
	}

	render() {
		return (
			<div style={styles.container}>
				<h1>Feedback system</h1>
				<Input type="input" label="Name" />
				<Input type="email" label="Email" />
				<HomeButton />
				<StarRating />
			</div>
		);
	}
}
export default withRouter(Home);
