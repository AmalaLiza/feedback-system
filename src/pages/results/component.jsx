import React from "react";
import PropTypes from "prop-types";
import { withRouter } from "react-router-dom";
import styles from "./styles.css";
import AreaGraph from "../../components/area-graph/component";
import BarGraph from "../../components/bar-graph/component";
import Title from "../../components/title/component";
import CommentBox from "../../components/comment-box/component";
import transformUtil from "../../utils/transform-data";

/**
 * Results component to render comments
 * CommentBox is used to render each comment.
 * @param {object} comments
 */
export const Results = ({ comments = [] }) => (
	<div className={styles.container}>
		<Title title="COMMENTS AND TRENDS" />
		<div className={styles.wrapper}>
			<ul className={styles.commentWrapper}>
				{comments?.map((comment) => (
					<CommentBox key={comment.id} comment={comment} />
				))}
			</ul>
			<div>
				<div className={styles.graphWrapper}>
					<AreaGraph
						chartData={transformUtil.getReadableDate(comments)}
						xAxisDataKey="commentDate"
						areaDataKey="rating"
					/>
					<span className={styles.description}>User rating/day</span>
				</div>
				<div className={styles.graphWrapper}>
					<BarGraph
						chartData={transformUtil.getTransformChartData(comments)}
						xAxisDataKey="rating"
						barDataKey="count"
					/>
					<span className={styles.description}>Rating Frequency</span>
				</div>
			</div>
		</div>
	</div>
);

Results.propTypes = {
	/**
	 * comments: comments to be rendered in results page
	 */
	comments: PropTypes.arrayOf(
		PropTypes.shape({
			email: PropTypes.string,
			name: PropTypes.string,
			comment: PropTypes.string,
			rating: PropTypes.number,
			commentDate: PropTypes.number,
		})
	),
};
export default withRouter(Results);
