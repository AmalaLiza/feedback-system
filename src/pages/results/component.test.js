import React from "react";
import { shallow } from "enzyme";
import { Results } from "./component";
import renderer from "react-test-renderer";
import BarGraph from "../../components/bar-graph/component";
import AreaGraph from "../../components/area-graph/component";
import CommentBox from "../../components/comment-box/component";

const props = {
	comments: [
		{
			id: 1,
			name: "Sheldon Cooper",
			email: "jonaskahnwald@dark.anotherworld",
			comment: "The end is the beginning, and the beginning is the end.",
			rating: 5,
		},
		{
			id: 2,
			name: "Sheldon Cooper",
			email: "jonaskahnwald@dark.anotherworld",
			comment: "The end is the beginning, and the beginning is the end.",
			rating: 4,
		},
	],
};

function setup() {
	const wrapper = shallow(<Results {...props} />);
	return { wrapper, props };
}

describe("Results Component Test Suite", () => {
	it("renders correctly", () => {
		const tree = renderer.create(<Results {...props} />).toJSON();
		expect(tree).toMatchSnapshot();
	});

	it("Should have an 1 Area Chart and 1 Bar Chart", () => {
		const { wrapper } = setup();
		expect(wrapper.find(BarGraph).exists()).toBe(true);
		expect(wrapper.find(BarGraph).length).toBe(1);
		expect(wrapper.find(AreaGraph).exists()).toBe(true);
		expect(wrapper.find(AreaGraph).length).toBe(1);
	});

	it("Should have an CommentBox length same as length of prop comments", () => {
		const { wrapper } = setup();
		expect(wrapper.find(CommentBox).exists()).toBe(true);
		expect(wrapper.find(CommentBox).length).toBe(props.comments.length);
	});
});
