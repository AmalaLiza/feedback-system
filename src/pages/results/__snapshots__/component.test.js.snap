// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Results Component Test Suite renders correctly 1`] = `
<div
  className="container"
>
  <h1
    className="title"
  >
    COMMENTS AND TRENDS
  </h1>
  <div
    className="wrapper"
  >
    <ul
      className="commentWrapper"
    >
      <li
        className="commentSection"
      >
        <header
          className="header"
        >
          <div
            className="avatar"
          >
            S
          </div>
          <span
            className="name"
          >
            Sheldon Cooper
          </span>
          <div
            className="metadata"
          >
            <div
              className="commentDate"
            >
              Invalid Date
            </div>
            <div
              className="commentDate"
            >
              <img
                className="star"
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
              5
            </div>
          </div>
        </header>
        <div
          className="commentContent"
        >
          <p
            className="email"
          >
            <span
              className="emailLabel"
            >
              E-mail 
            </span>
            <span>
              jonaskahnwald@dark.anotherworld
            </span>
          </p>
          <p>
            The end is the beginning, and the beginning is the end.
          </p>
        </div>
      </li>
      <li
        className="commentSection"
      >
        <header
          className="header"
        >
          <div
            className="avatar"
          >
            S
          </div>
          <span
            className="name"
          >
            Sheldon Cooper
          </span>
          <div
            className="metadata"
          >
            <div
              className="commentDate"
            >
              Invalid Date
            </div>
            <div
              className="commentDate"
            >
              <img
                className="star"
                src={
                  Object {
                    "process": [Function],
                  }
                }
              />
              4
            </div>
          </div>
        </header>
        <div
          className="commentContent"
        >
          <p
            className="email"
          >
            <span
              className="emailLabel"
            >
              E-mail 
            </span>
            <span>
              jonaskahnwald@dark.anotherworld
            </span>
          </p>
          <p>
            The end is the beginning, and the beginning is the end.
          </p>
        </div>
      </li>
    </ul>
    <div>
      <div
        className="graphWrapper"
      >
        <div
          className="recharts-wrapper"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseMove={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "cursor": "default",
              "height": 400,
              "position": "relative",
              "width": 500,
            }
          }
        >
          <svg
            className="recharts-surface"
            fillOpacity={1}
            height={400}
            version="1.1"
            viewBox="0 0 500 400"
            width={500}
          >
            <defs>
              <clipPath
                id="recharts2-clip"
              >
                <rect
                  height={360}
                  width={410}
                  x={60}
                  y={10}
                />
              </clipPath>
            </defs>
            <g
              className="recharts-cartesian-grid"
            >
              <g
                className="recharts-cartesian-grid-horizontal"
              >
                <line
                  fill="none"
                  height={360}
                  stroke="#ccc"
                  strokeDasharray="3 3"
                  width={410}
                  x={60}
                  x1={60}
                  x2={470}
                  y={10}
                  y1={370}
                  y2={370}
                />
                <line
                  fill="none"
                  height={360}
                  stroke="#ccc"
                  strokeDasharray="3 3"
                  width={410}
                  x={60}
                  x1={60}
                  x2={470}
                  y={10}
                  y1={298}
                  y2={298}
                />
                <line
                  fill="none"
                  height={360}
                  stroke="#ccc"
                  strokeDasharray="3 3"
                  width={410}
                  x={60}
                  x1={60}
                  x2={470}
                  y={10}
                  y1={226}
                  y2={226}
                />
                <line
                  fill="none"
                  height={360}
                  stroke="#ccc"
                  strokeDasharray="3 3"
                  width={410}
                  x={60}
                  x1={60}
                  x2={470}
                  y={10}
                  y1={154}
                  y2={154}
                />
                <line
                  fill="none"
                  height={360}
                  stroke="#ccc"
                  strokeDasharray="3 3"
                  width={410}
                  x={60}
                  x1={60}
                  x2={470}
                  y={10}
                  y1={81.99999999999999}
                  y2={81.99999999999999}
                />
                <line
                  fill="none"
                  height={360}
                  stroke="#ccc"
                  strokeDasharray="3 3"
                  width={410}
                  x={60}
                  x1={60}
                  x2={470}
                  y={10}
                  y1={10}
                  y2={10}
                />
              </g>
              <g
                className="recharts-cartesian-grid-vertical"
              >
                <line
                  fill="none"
                  height={360}
                  stroke="#ccc"
                  strokeDasharray="3 3"
                  width={410}
                  x={60}
                  x1={60}
                  x2={60}
                  y={10}
                  y1={10}
                  y2={370}
                />
                <line
                  fill="none"
                  height={360}
                  stroke="#ccc"
                  strokeDasharray="3 3"
                  width={410}
                  x={60}
                  x1={470}
                  x2={470}
                  y={10}
                  y1={10}
                  y2={370}
                />
              </g>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
            >
              <line
                className="recharts-cartesian-axis-line"
                fill="none"
                height={30}
                stroke="#666"
                width={410}
                x={60}
                x1={60}
                x2={470}
                y={370}
                y1={370}
                y2={370}
              />
              <g
                className="recharts-cartesian-axis-ticks"
              >
                <g
                  className="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    className="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height={30}
                    stroke="#666"
                    width={410}
                    x={60}
                    x1={60}
                    x2={60}
                    y={370}
                    y1={376}
                    y2={370}
                  />
                  <text
                    className="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height={30}
                    stroke="none"
                    textAnchor="middle"
                    width={410}
                    x={60}
                    y={378}
                  >
                    <tspan
                      dy="0.71em"
                      x={60}
                    >
                      Invalid Date
                    </tspan>
                  </text>
                </g>
                <g
                  className="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    className="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height={30}
                    stroke="#666"
                    width={410}
                    x={60}
                    x1={470}
                    x2={470}
                    y={370}
                    y1={376}
                    y2={370}
                  />
                  <text
                    className="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height={30}
                    stroke="none"
                    textAnchor="middle"
                    width={410}
                    x={470}
                    y={378}
                  >
                    <tspan
                      dy="0.71em"
                      x={470}
                    >
                      Invalid Date
                    </tspan>
                  </text>
                </g>
              </g>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
            >
              <line
                className="recharts-cartesian-axis-line"
                fill="none"
                height={360}
                stroke="#666"
                width={60}
                x={0}
                x1={60}
                x2={60}
                y={10}
                y1={10}
                y2={370}
              />
              <g
                className="recharts-cartesian-axis-ticks"
              >
                <g
                  className="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    className="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height={360}
                    stroke="#666"
                    width={60}
                    x={0}
                    x1={54}
                    x2={60}
                    y={10}
                    y1={370}
                    y2={370}
                  />
                  <text
                    className="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height={360}
                    stroke="none"
                    textAnchor="end"
                    width={60}
                    x={52}
                    y={370}
                  >
                    <tspan
                      dy="0.355em"
                      x={52}
                    >
                      0
                    </tspan>
                  </text>
                </g>
                <g
                  className="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    className="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height={360}
                    stroke="#666"
                    width={60}
                    x={0}
                    x1={54}
                    x2={60}
                    y={10}
                    y1={298}
                    y2={298}
                  />
                  <text
                    className="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height={360}
                    stroke="none"
                    textAnchor="end"
                    width={60}
                    x={52}
                    y={298}
                  >
                    <tspan
                      dy="0.355em"
                      x={52}
                    >
                      1
                    </tspan>
                  </text>
                </g>
                <g
                  className="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    className="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height={360}
                    stroke="#666"
                    width={60}
                    x={0}
                    x1={54}
                    x2={60}
                    y={10}
                    y1={226}
                    y2={226}
                  />
                  <text
                    className="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height={360}
                    stroke="none"
                    textAnchor="end"
                    width={60}
                    x={52}
                    y={226}
                  >
                    <tspan
                      dy="0.355em"
                      x={52}
                    >
                      2
                    </tspan>
                  </text>
                </g>
                <g
                  className="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    className="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height={360}
                    stroke="#666"
                    width={60}
                    x={0}
                    x1={54}
                    x2={60}
                    y={10}
                    y1={154}
                    y2={154}
                  />
                  <text
                    className="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height={360}
                    stroke="none"
                    textAnchor="end"
                    width={60}
                    x={52}
                    y={154}
                  >
                    <tspan
                      dy="0.355em"
                      x={52}
                    >
                      3
                    </tspan>
                  </text>
                </g>
                <g
                  className="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    className="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height={360}
                    stroke="#666"
                    width={60}
                    x={0}
                    x1={54}
                    x2={60}
                    y={10}
                    y1={81.99999999999999}
                    y2={81.99999999999999}
                  />
                  <text
                    className="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height={360}
                    stroke="none"
                    textAnchor="end"
                    width={60}
                    x={52}
                    y={81.99999999999999}
                  >
                    <tspan
                      dy="0.355em"
                      x={52}
                    >
                      4
                    </tspan>
                  </text>
                </g>
                <g
                  className="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    className="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height={360}
                    stroke="#666"
                    width={60}
                    x={0}
                    x1={54}
                    x2={60}
                    y={10}
                    y1={10}
                    y2={10}
                  />
                  <text
                    className="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height={360}
                    stroke="none"
                    textAnchor="end"
                    width={60}
                    x={52}
                    y={10}
                  >
                    <tspan
                      dy="0.355em"
                      x={52}
                    >
                      5
                    </tspan>
                  </text>
                </g>
              </g>
            </g>
            <g
              className="recharts-layer recharts-area"
            >
              <g
                className="recharts-layer"
              >
                <defs>
                  <clipPath
                    id="animationClipPath-recharts-area-3"
                  >
                    <rect
                      height={371}
                      width={0}
                      x={60}
                      y={0}
                    />
                  </clipPath>
                </defs>
                <g
                  className="recharts-layer"
                  clipPath="url(#animationClipPath-recharts-area-3)"
                >
                  <g
                    className="recharts-layer"
                    clipPath={null}
                  >
                    <path
                      className="recharts-curve recharts-area-area"
                      d="M60,10L470,81.99999999999999L470,370L60,370Z"
                      fill="url(#colorUv)"
                      fillOpacity={0.8}
                      height={360}
                      stroke="none"
                      width={410}
                    />
                    <path
                      className="recharts-curve recharts-area-curve"
                      d="M60,10L470,81.99999999999999"
                      fill="none"
                      fillOpacity={0.8}
                      height={360}
                      stroke="#f15c41"
                      width={410}
                    />
                  </g>
                </g>
              </g>
            </g>
            <defs>
              <linearGradient
                id="colorUv"
                x1="0"
                x2="0"
                y1="0"
                y2="1"
              >
                <stop
                  offset="5%"
                  stopColor="#e052a0"
                  stopOpacity={1}
                />
                <stop
                  fillOpacity={1}
                  offset="95%"
                  stopColor="#f15c41"
                  stopOpacity={1}
                />
              </linearGradient>
            </defs>
          </svg>
          <div
            className="recharts-tooltip-wrapper"
            style={
              Object {
                "MozTransform": "translate(undefinedpx, undefinedpx)",
                "OTransform": "translate(undefinedpx, undefinedpx)",
                "WebkitTransform": "translate(undefinedpx, undefinedpx)",
                "msTransform": "translate(undefinedpx, undefinedpx)",
                "pointerEvents": "none",
                "position": "absolute",
                "top": 0,
                "transform": "translate(undefinedpx, undefinedpx)",
                "visibility": "hidden",
              }
            }
          />
        </div>
        <span
          className="description"
        >
          User rating/day
        </span>
      </div>
      <div
        className="graphWrapper"
      >
        <div
          className="recharts-wrapper"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseMove={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          style={
            Object {
              "cursor": "default",
              "height": 400,
              "position": "relative",
              "width": 500,
            }
          }
        >
          <svg
            className="recharts-surface"
            fillOpacity={1}
            height={400}
            version="1.1"
            viewBox="0 0 500 400"
            width={500}
          >
            <defs>
              <clipPath
                id="recharts5-clip"
              >
                <rect
                  height={360}
                  width={410}
                  x={60}
                  y={10}
                />
              </clipPath>
            </defs>
            <g
              className="recharts-cartesian-grid"
            >
              <g
                className="recharts-cartesian-grid-horizontal"
              >
                <line
                  fill="none"
                  height={360}
                  stroke="#ccc"
                  strokeDasharray="3 3"
                  width={410}
                  x={60}
                  x1={60}
                  x2={470}
                  y={10}
                  y1={370}
                  y2={370}
                />
                <line
                  fill="none"
                  height={360}
                  stroke="#ccc"
                  strokeDasharray="3 3"
                  width={410}
                  x={60}
                  x1={60}
                  x2={470}
                  y={10}
                  y1={298}
                  y2={298}
                />
                <line
                  fill="none"
                  height={360}
                  stroke="#ccc"
                  strokeDasharray="3 3"
                  width={410}
                  x={60}
                  x1={60}
                  x2={470}
                  y={10}
                  y1={226}
                  y2={226}
                />
                <line
                  fill="none"
                  height={360}
                  stroke="#ccc"
                  strokeDasharray="3 3"
                  width={410}
                  x={60}
                  x1={60}
                  x2={470}
                  y={10}
                  y1={154}
                  y2={154}
                />
                <line
                  fill="none"
                  height={360}
                  stroke="#ccc"
                  strokeDasharray="3 3"
                  width={410}
                  x={60}
                  x1={60}
                  x2={470}
                  y={10}
                  y1={81.99999999999999}
                  y2={81.99999999999999}
                />
                <line
                  fill="none"
                  height={360}
                  stroke="#ccc"
                  strokeDasharray="3 3"
                  width={410}
                  x={60}
                  x1={60}
                  x2={470}
                  y={10}
                  y1={10}
                  y2={10}
                />
              </g>
              <g
                className="recharts-cartesian-grid-vertical"
              >
                <line
                  fill="none"
                  height={360}
                  stroke="#ccc"
                  strokeDasharray="3 3"
                  width={410}
                  x={60}
                  x1={162.5}
                  x2={162.5}
                  y={10}
                  y1={10}
                  y2={370}
                />
                <line
                  fill="none"
                  height={360}
                  stroke="#ccc"
                  strokeDasharray="3 3"
                  width={410}
                  x={60}
                  x1={367.5}
                  x2={367.5}
                  y={10}
                  y1={10}
                  y2={370}
                />
                <line
                  fill="none"
                  height={360}
                  stroke="#ccc"
                  strokeDasharray="3 3"
                  width={410}
                  x={60}
                  x1={60}
                  x2={60}
                  y={10}
                  y1={10}
                  y2={370}
                />
                <line
                  fill="none"
                  height={360}
                  stroke="#ccc"
                  strokeDasharray="3 3"
                  width={410}
                  x={60}
                  x1={470}
                  x2={470}
                  y={10}
                  y1={10}
                  y2={370}
                />
              </g>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis recharts-xAxis xAxis"
            >
              <line
                className="recharts-cartesian-axis-line"
                fill="none"
                height={30}
                stroke="#666"
                width={410}
                x={60}
                x1={60}
                x2={470}
                y={370}
                y1={370}
                y2={370}
              />
              <g
                className="recharts-cartesian-axis-ticks"
              >
                <g
                  className="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    className="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height={30}
                    stroke="#666"
                    width={410}
                    x={60}
                    x1={162.5}
                    x2={162.5}
                    y={370}
                    y1={376}
                    y2={370}
                  />
                  <text
                    className="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height={30}
                    stroke="none"
                    textAnchor="middle"
                    width={410}
                    x={162.5}
                    y={378}
                  >
                    <tspan
                      dy="0.71em"
                      x={162.5}
                    >
                      4
                    </tspan>
                  </text>
                </g>
                <g
                  className="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    className="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height={30}
                    stroke="#666"
                    width={410}
                    x={60}
                    x1={367.5}
                    x2={367.5}
                    y={370}
                    y1={376}
                    y2={370}
                  />
                  <text
                    className="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height={30}
                    stroke="none"
                    textAnchor="middle"
                    width={410}
                    x={367.5}
                    y={378}
                  >
                    <tspan
                      dy="0.71em"
                      x={367.5}
                    >
                      5
                    </tspan>
                  </text>
                </g>
              </g>
            </g>
            <g
              className="recharts-layer recharts-cartesian-axis recharts-yAxis yAxis"
            >
              <line
                className="recharts-cartesian-axis-line"
                fill="none"
                height={360}
                stroke="#666"
                width={60}
                x={0}
                x1={60}
                x2={60}
                y={10}
                y1={10}
                y2={370}
              />
              <g
                className="recharts-cartesian-axis-ticks"
              >
                <g
                  className="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    className="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height={360}
                    stroke="#666"
                    width={60}
                    x={0}
                    x1={54}
                    x2={60}
                    y={10}
                    y1={370}
                    y2={370}
                  />
                  <text
                    className="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height={360}
                    stroke="none"
                    textAnchor="end"
                    width={60}
                    x={52}
                    y={370}
                  >
                    <tspan
                      dy="0.355em"
                      x={52}
                    >
                      0
                    </tspan>
                  </text>
                </g>
                <g
                  className="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    className="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height={360}
                    stroke="#666"
                    width={60}
                    x={0}
                    x1={54}
                    x2={60}
                    y={10}
                    y1={298}
                    y2={298}
                  />
                  <text
                    className="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height={360}
                    stroke="none"
                    textAnchor="end"
                    width={60}
                    x={52}
                    y={298}
                  >
                    <tspan
                      dy="0.355em"
                      x={52}
                    >
                      1
                    </tspan>
                  </text>
                </g>
                <g
                  className="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    className="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height={360}
                    stroke="#666"
                    width={60}
                    x={0}
                    x1={54}
                    x2={60}
                    y={10}
                    y1={226}
                    y2={226}
                  />
                  <text
                    className="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height={360}
                    stroke="none"
                    textAnchor="end"
                    width={60}
                    x={52}
                    y={226}
                  >
                    <tspan
                      dy="0.355em"
                      x={52}
                    >
                      2
                    </tspan>
                  </text>
                </g>
                <g
                  className="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    className="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height={360}
                    stroke="#666"
                    width={60}
                    x={0}
                    x1={54}
                    x2={60}
                    y={10}
                    y1={154}
                    y2={154}
                  />
                  <text
                    className="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height={360}
                    stroke="none"
                    textAnchor="end"
                    width={60}
                    x={52}
                    y={154}
                  >
                    <tspan
                      dy="0.355em"
                      x={52}
                    >
                      3
                    </tspan>
                  </text>
                </g>
                <g
                  className="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    className="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height={360}
                    stroke="#666"
                    width={60}
                    x={0}
                    x1={54}
                    x2={60}
                    y={10}
                    y1={81.99999999999999}
                    y2={81.99999999999999}
                  />
                  <text
                    className="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height={360}
                    stroke="none"
                    textAnchor="end"
                    width={60}
                    x={52}
                    y={81.99999999999999}
                  >
                    <tspan
                      dy="0.355em"
                      x={52}
                    >
                      4
                    </tspan>
                  </text>
                </g>
                <g
                  className="recharts-layer recharts-cartesian-axis-tick"
                >
                  <line
                    className="recharts-cartesian-axis-tick-line"
                    fill="none"
                    height={360}
                    stroke="#666"
                    width={60}
                    x={0}
                    x1={54}
                    x2={60}
                    y={10}
                    y1={10}
                    y2={10}
                  />
                  <text
                    className="recharts-text recharts-cartesian-axis-tick-value"
                    fill="#666"
                    height={360}
                    stroke="none"
                    textAnchor="end"
                    width={60}
                    x={52}
                    y={10}
                  >
                    <tspan
                      dy="0.355em"
                      x={52}
                    >
                      5
                    </tspan>
                  </text>
                </g>
              </g>
            </g>
            <g
              className="recharts-layer recharts-bar"
            >
              <g
                className="recharts-layer recharts-bar-rectangles"
                clipPath={null}
              >
                <g
                  className="recharts-layer"
                >
                  <g
                    className="recharts-layer recharts-bar-rectangle"
                  />
                  <g
                    className="recharts-layer recharts-bar-rectangle"
                  />
                </g>
              </g>
            </g>
            <defs>
              <linearGradient
                id="colorUv"
                x1="0"
                x2="0"
                y1="0"
                y2="1"
              >
                <stop
                  offset="5%"
                  stopColor="#e052a0"
                  stopOpacity={1}
                />
                <stop
                  fillOpacity={1}
                  offset="95%"
                  stopColor="#f15c41"
                  stopOpacity={1}
                />
              </linearGradient>
            </defs>
          </svg>
          <div
            className="recharts-tooltip-wrapper"
            style={
              Object {
                "MozTransform": "translate(undefinedpx, undefinedpx)",
                "OTransform": "translate(undefinedpx, undefinedpx)",
                "WebkitTransform": "translate(undefinedpx, undefinedpx)",
                "msTransform": "translate(undefinedpx, undefinedpx)",
                "pointerEvents": "none",
                "position": "absolute",
                "top": 0,
                "transform": "translate(undefinedpx, undefinedpx)",
                "visibility": "hidden",
              }
            }
          />
        </div>
        <span
          className="description"
        >
          Rating Frequency
        </span>
      </div>
    </div>
  </div>
</div>
`;
